src/main/java/at/hannibal2/skyhanni/features/misc/EmojiReplacer.kt

```diff
@@ -0,0 +1,206 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.api.event.HandleEvent
+import at.hannibal2.skyhanni.data.jsonobjects.repo.EmojiJson
+import at.hannibal2.skyhanni.events.RepositoryReloadEvent
+import at.hannibal2.skyhanni.mixins.transformers.AccessorMixinGuiChat
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import net.minecraft.client.gui.GuiChat
+import net.minecraft.client.gui.GuiScreen
+import net.minecraft.client.renderer.texture.TextureManager
+import net.minecraft.util.ResourceLocation
+import org.lwjgl.input.Keyboard
+import org.lwjgl.opengl.GL11
+import org.lwjgl.util.Color
+
+@SkyHanniModule
+object EmojiReplacer {
+    private val config get() = SkyHanniMod.feature.gui
+    private var emojiNameMap: Map<String, Int>? = null
+    private var reverseUnicodeToName: Map<String, String>? = null
+    @HandleEvent
+    fun onRepoReload(event: RepositoryReloadEvent) {
+        val emojiJson = event.getConstant<EmojiJson>("Emojis")
+        emojiNameMap = emojiJson.emojiNames
+        reverseUnicodeToName = emojiJson.unicodeNames
+    }
+    private var stringIndex: Int = -1
+    private var emojiEnd = -1
+    private var renderedString: String = ""
+    private var isShadow = false
+    private val emojiResource = ResourceLocation("skyhanni:emoji/emoji_table.png")
```

Comment:
there is some compat function to make resource locations, Im pretty sure its merged im beta